app/helpers/validation/profile.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+import logging
+
+import geojson
+from flask import abort
+from shapely.geometry import shape
+
+from app import make_error_msg
+from app.helpers.profile_helpers import PROFILE_MAX_AMOUNT_POINTS, PROFILE_DEFAULT_AMOUNT_POINTS
+from app.helpers.validation import srs_guesser
+
+logger = logging.getLogger(__name__)
+
+
+def read_linestring(request_object):
+    # param geom, list of coordinates defining the line on which we want a profile
+    linestring = None
+    if 'geom' in request_object.args:
+        linestring = request_object.args.get('geom')
+    elif request_object.is_json and len(request_object.get_data(as_text=True)) > 0:
+        linestring = request_object.get_json()
+    if not linestring:
+        abort(400, "No 'geom' given, cannot create a profile without coordinates")
+    try:
+        geom = geojson.loads(linestring, object_hook=geojson.GeoJSON.to_instance)
+    except ValueError:
+        raise abort(400, "Error loading geometry in JSON string")
+    try:
+        geom_to_shape = shape(geom)
+    except Exception as e:
+        logger.exception(e)
+        raise abort(400, "Error converting JSON to Shape")
```

Comment:
I would avoid broad exception here. Broad Exception are already caught by flask returning 500. If we know that some exception might happen we can leave it with the logger.exception and re-raise the exception

Reply:
As it is a matter of validating what the user is giving us, I thought it would be more fitting to reject the request with a 40x if user's input are not valid instead of failing with a 50x (which looks like a badly implemented backend IMO, 500 is like NullPointerException in Java, you shouldn't experience them other than in dev phase)

Reply:
OK then I suggest to simply remove the raise before abort and give a proper error message in the logger.exception